File: fixtures/scale/kotlin/medium/src/calc/funcs/Powers.kt
package calc.funcs

import calc.eval.FunctionRegistry
import kotlin.math.cbrt
import kotlin.math.hypot
import kotlin.math.pow
import kotlin.math.sqrt

fun registerPowers(registry: FunctionRegistry) {
    registry.addUnary("sqrt") { sqrt(it) }
    registry.addUnary("cbrt") { cbrt(it) }
    registry.addUnary("square") { it * it }
    registry.addAggregate("hypot") { hypot(it[0], it[1]) }
    registry.addAggregate("pow") { it[0].pow(it[1]) }
}

File: fixtures/scale/scala/medium/src/calc/funcs/Powers.scala
package calc.funcs

import calc.eval.FunctionRegistry

object Powers {
  def register(registry: FunctionRegistry): Unit = {
    registry.addUnary("sqrt", math.sqrt)
    registry.addUnary("cbrt", math.cbrt)
    registry.addUnary("square", x => x * x)
    registry.addAggregate("hypot", args => math.hypot(args(0), args(1)))
    registry.addAggregate("pow", args => math.pow(args(0), args(1)))
  }
}
